### ISSUE TITLE
Request: Return actual NA_character_ entries on the component functions

### ISSUE THREAD

**Issue by alexcpsec** [Ironholds/urltools#34]
When I try the function with a simple NA_character_ , it is returning a `"na"` string. 

I think it would be better if it preserved the NA_character_ to make a workflow using it more consistent.

````
> domain(NA_character_)
[1] "na"
````

**Comment by Ironholds** [comment]
>#include <Rcpp.h>
using namespace Rcpp;

>// [[Rcpp::export]]
std::string test(std::string x) {
  return x;
}

>test(NA_character_)

>[1] "NA"
</code>

strings (and vectors of strings) can't handle NAs: only R's custom object types (CharacterVector, in this case) can. Switching over to them would involve significant refactoring and tremendously impact speed, since we'd be unable to rely on a string as the base unit, which means the current internal architecture, which is oriented around the repeated tweaking of a string reference, wouldn't work.

Basically in order to make this workflow more consistent you're looking at a lot of refactoring, unless we do something dumb like a consistent check for if the output is "na", which would slow things down...a lot.

**Comment by alexcpsec** [comment]
Ok, no worries. I can code around it, but it really caught me by surprise there.

Maybe a note on the docs that this is the expected behavior then?

**Comment by Ironholds** [comment]
Yeah, makes sense; I'll plug it into the next version :)
